server/src/main/java/io/spine/server/aggregate/AggregateStorage.java

```diff
@@ -204,4 +208,52 @@ private static AggregateEventRecord toStorageRecord(Snapshot snapshot) {
      */
     protected abstract Iterator<AggregateEventRecord> historyBackward(
             AggregateReadRequest<I> request);
+
+    /**
+     * Drops all records which occur before the Nth snapshot for each entity.
+     *
+     * <p>The snapshot number is counted from the latest to earliest, where {@code 1} represents
+     * the latest snapshot.
+     *
+     * <p>The snapshot number higher than the overall snapshot count of the entity is allowed, the
+     * entity records remain intact in this case.
+     *
+     * @throws IllegalArgumentException
+     *         if the {@code snapshotNumber} is {@code 0} or less
+     */
+    @Internal
+    public void clipRecordsBeforeSnapshot(int snapshotNumber) {
```

Comment:
`truncateOlderThan(int snapshotIndex)`
and the one below will be
`truncateOlderThan(int snapshotIndex, Timestamp date)`

Please update the naming of the `protected` methods and the documentation accordingly.